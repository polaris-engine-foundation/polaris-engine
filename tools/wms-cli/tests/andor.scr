func main() {
    a = 5;
    if (a > 0 && a < 10) {
        print("1 true");
    }
    if (a > 0 && a < 9) {
        print("2 true");
    }
    if (a > 0 && a < 8) {
        print("3 true");
    }
    if (a > 0 && a < 7) {
        print("4 true");
    }
    if (a > 0 && a < 6) {
        print("5 true");
    }
    if (a > 0 && a < 5) {
        print("6 false");
    }
    if (a > 0 && a < 4) {
        print("7 false");
    }

    print("--");

    a = 5;
    if (a > 0 && a < 10) {
        print("1 true");
    }
    if (a > 1 && a < 10) {
        print("2 true");
    }
    if (a > 2 && a < 10) {
        print("3 true");
    }
    if (a > 3 && a < 10) {
        print("4 true");
    }
    if (a > 4 && a < 10) {
        print("5 true");
    }
    if (a > 5 && a < 10) {
        print("6 false");
    }
    if (a > 6 && a < 10) {
        print("7 false");
    }

    print("--");

    a = 1;
    b = 2;
    if (a == 1 && b == 0) {
        print("1 false");
    }
    if (a == 0 && b == 2) {
        print("2 false");
    }
    if (a == 1 && b == 2) {
        print("3 true");
    }
    if (a == 0 && b == 0) {
        print("4 false");
    }

    print("--");

    a = 1;
    b = 2;
    if (a == 1 || b == 0) {
        print("1 true");
    }
    if (a == 0 || b == 2) {
        print("2 true");
    }
    if (a == 1 || b == 2) {
        print("3 true");
    }
    if (a == 0 || b == 0) {
        print("4 false");
    }

}
