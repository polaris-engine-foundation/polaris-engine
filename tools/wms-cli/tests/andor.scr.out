1 true
2 true
3 true
4 true
5 true
--
1 true
2 true
3 true
4 true
5 true
--
3 true
--
1 true
2 true
3 true
